apps/riscv-tests/isa/macros/vector/vector_macros.h

```diff
@@ -126,11 +127,20 @@ int test_case;
   printf("PASSED.\n");
 
 // Macros to set vector length, type and multiplier
+// Don't use this to set VL == 0 since the compiler puts rs1 == x0
 #define VSET(VLEN,VTYPE,LMUL)                                                          \
   do {                                                                                 \
   asm volatile ("vsetvli t0, %[A]," #VTYPE "," #LMUL ", ta, ma \n" :: [A] "r" (VLEN)); \
   } while(0)
 
+// Macros to set vector length equal to zero
+#define VSET_ZERO(VTYPE,LMUL)                                                              \
+  do {                                                                                     \
+    int vset_zero_buf;                                                                     \
+    asm volatile("li %0, 0" : "=r" (vset_zero_buf));                                       \
+    asm volatile("vsetvli x0, %0," #VTYPE "," #LMUL ", ta, ma \n" :: "r" (vset_zero_buf)); \
+  } while(0)
```

Comment:
Can we combine those in a single macro? One which (hopefully) would be optimized at compile-time?

Reply:
Problem is that it should not be too optimized :P 
`vsetvli x0, x0` would leave the same `vl` already in place; the second register should contain `0` without being `x0`

Reply:
Yeah, I thought so, but I do not like the duplicated macros.. Oh well